AzureTable/Microsoft.DataTransfer.AzureTable/Shared/IAzureTableAdapterConfiguration.cs

```diff
@@ -12,5 +13,11 @@ public interface IAzureTableAdapterConfiguration
         /// </summary>
         [Display(ResourceType = typeof(ConfigurationResources), Description = "ConnectionString")]
         string ConnectionString { get; }
+
+        /// <summary>
+        /// Gets the value that indicates which location mode to use when accessing data from azure tables.
+        /// </summary>
+        [Display(ResourceType = typeof(DynamicConfigurationResources), Description = "Source_LocationMode")]
+        AzureTableLocationMode LocationMode { get; }
```

Comment:
Actually, in order for this one to be optional - I think you will need to make it nullable. This will also make it consistent with Azure Storage SDK, so that we can pass `null` in there to use their default.